externalId-migration-util/src/main/java/org/sunbird/migration/RecordProcessor.java

```diff
@@ -0,0 +1,249 @@
+        ResultSet resultSet = connection.getRecords("select * from organisation");
+        Map<String,String> orgProviderMap = CassandraHelper.getOrgProviderMapFromResultSet(resultSet);
+        logger.debug(orgProviderMap.entrySet().iterator().next());
+        return orgProviderMap;
+    }
+
+    /**
+     * this method is responsible to fetch the user external id data from cassandra and the table name
+     * is usr_external_identity and after fetching will convert the resultSet to List<User>
+     *
+     * @return List<User>
+     */
+    private List<User> getUserDataFromDbAsList() {
+        ResultSet resultSet = connection.getRecords("select * from usr_external_identity");
+        List<User> usersList = CassandraHelper.getUserListFromResultSet(resultSet);
+        logger.debug(usersList.iterator().next());
+        return usersList;
+    }
+
+    /**
+     * this method will be used to process externalId this method will get the count of total number
+     * of records that need to be processed
+     *
+     * @return
+     */
+    public void startProcessingExternalIds() throws IOException {
+        List<User> usersList = getRecordsToBeProcessed();
+        Map<String,String> orgIdProviderMap = getOrgDataFromDbAsList();
+        Map<String,List<User>> userIdExternalIdMap = createExternalInfoMapGroupbyUserId(usersList);
+
+        List<UserDeclareEntity> userSelfDeclareLists = new ArrayList<>();
+        getUserSelfDeclareLists(userIdExternalIdMap, userSelfDeclareLists);
+        userSelfDeclareLists
+                .stream()
+                .forEach(
+                        userSelfDeclareObject->{
+                            try{
+                                startTracingRecord(userSelfDeclareObject.getUserId());
+                                performSequentialOperationOnRecord(userSelfDeclareObject,orgIdProviderMap);
+                            }catch (Exception ex){
+                                logExceptionOnProcessingRecord(userSelfDeclareObject.getUserId(),userSelfDeclareObject.getProvider(),
+                                        userSelfDeclareObject.getPersona());
+                            }finally {
+                                endTracingRecord(userSelfDeclareObject.getUserId());
+                            }
+                 });
+        connection.closeConnection();
+        closeWriterConnection();
+
+    }
+
+    private void getUserSelfDeclareLists(Map<String, List<User>> userIdExternalIdMap, List<UserDeclareEntity> userSelfDeclareLists) {
+        for (Map.Entry<String,List<User>> userEntry: userIdExternalIdMap.entrySet()){
+            String userId = userEntry.getKey();
+            List<User> users = userEntry.getValue();
+            UserDeclareEntity userDeclareEntity = new UserDeclareEntity();
+            userDeclareEntity.setUserId(userId);
+            userDeclareEntity.setProvider(users.get(0).getOriginalProvider());
+            userDeclareEntity.setPersona(Constants.TEACHER);
```

Comment:
we need to ensure this string is same as what portal is going to decide. just saying it aloud here.

Reply:
I am assuming we will have only one provider info for the userId.. please let me know if there are chances of multiple provider for the same userId